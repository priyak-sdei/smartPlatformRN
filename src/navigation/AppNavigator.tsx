import React from 'react';
import { View } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import AuthNavigator from './AuthNavigator';
import TabNavigator from './TabNavigator';
import { useAppSelector } from '../redux/hooks';

const RootStack = createStackNavigator();

const AppNavigator = () => {
  const email = useAppSelector(state => state.user.email);
  return (
    <View testID="app-navigator-root" style={{ flex: 1 }}>
      <NavigationContainer>
        <RootStack.Navigator screenOptions={{ headerShown: false }}>
          {email ? (
            <RootStack.Screen name="MainTabs" component={TabNavigator} />
          ) : (
            <RootStack.Screen name="Auth" component={AuthNavigator} />
          )}
        </RootStack.Navigator>
      </NavigationContainer>
    </View>
  );
};

export default AppNavigator;
