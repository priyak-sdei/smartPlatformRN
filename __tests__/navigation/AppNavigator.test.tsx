import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import AppNavigator from '../../src/navigation/AppNavigator';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import renderer, { act } from 'react-test-renderer';

beforeAll(() => {
    // Save the original console.error
    // @ts-ignore
    console._errorOriginal = console.error;
    jest.spyOn(console, 'error').mockImplementation((msg) => {
        if (
            typeof msg === 'string' &&
            msg.includes('inside a test was not wrapped in act')
        ) {
            return;
        }
        // @ts-ignore
        console._errorOriginal(msg);
    });
});

afterAll(() => {
    // @ts-ignore
    if (console._errorOriginal) {
        // @ts-ignore
        console.error = console._errorOriginal;
    }
});

afterEach(cleanup);

describe('AppNavigator', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <AppNavigator />
            </Provider>
        );
        await waitFor(() => {
            expect(getByTestId('app-navigator-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(
                <Provider store={store}>
                    <AppNavigator />
                </Provider>
            );
        });
        expect(tree!.toJSON()).toMatchSnapshot('AppNavigator component');
    });
}); 