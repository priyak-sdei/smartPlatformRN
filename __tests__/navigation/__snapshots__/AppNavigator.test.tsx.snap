// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigator matches snapshot: AppNavigator component 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="app-navigator-root"
/>
`;
